app/streamSelector/browser.js

```diff
@@ -1,13 +1,41 @@
 const { desktopCapturer, ipcRenderer } = require('electron');
-window.addEventListener('DOMContentLoaded', createSourceSelector());
+window.addEventListener('DOMContentLoaded', init());
 
-function createSourceSelector() {
+function init() {
 	return () => {
-		ipcRenderer.once('get-screensizes-response', (event, screens) => {
-			createPreviewScreen(screens);
-		});
-		ipcRenderer.send('get-screensizes-request');
-	};
+		if (process.env["XDG_SESSION_TYPE"] === "wayland")
+			initRequestSource();
```

Comment:
```suggestion
                         //Pipewire dialog already allows user to select screen/window so request directly to avoid prompting user multiple times to select screen
			initRequestSource(requestSingleScreenOrWindow);
```